pkg/client/connector/install.go

```diff
@@ -262,19 +262,31 @@ func (ki *installer) updateDeployment(c context.Context, env client.Env, current
 	return dep, err
 }
 
-func svcPortByName(svc *kates.Service, name string) []*kates.ServicePort {
+// svcPortByNameOrNumber iterates through a list of ports in a service and
+// only returns the ports that match the given nameOrNumber
+func svcPortByNameOrNumber(svc *kates.Service, nameOrNumber string) []*kates.ServicePort {
 	svcPorts := make([]*kates.ServicePort, 0)
 	ports := svc.Spec.Ports
 	for i := range ports {
 		port := &ports[i]
-		if name == "" || name == port.Name {
+		matchFound := false
+		// If no nameOrNumber has been specified, we include it
+		if nameOrNumber == "" {
+			matchFound = true
+		}
+		// If a nameOrNumber has been specified, we check if it matches the Name
+		// or Port number, and add it if either case is true
+		if nameOrNumber == port.Name || nameOrNumber == strconv.Itoa(int(port.Port)) {
```

Comment:
Part of me feels weird about using `strconv.Itoa` instead of going the other direction.  What if the user wants to pass the number on the CLI with a leading 0?

And the other part of me thinks that I'm being ridiculous.

You could use `k8s.io/apimachinery/pkg/util/validation.IsValidPortName()` and `strconv.Atoi()` to see which one the string is before you even look at the Service.